src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReader.cs

```diff
@@ -2,35 +2,105 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+#nullable enable
+using System.Collections.Generic;
+
 namespace System.Security.Cryptography.Encoding.Tests.Cbor
 {
-    internal ref partial struct CborValueReader
+    internal enum CborReaderState
+    {
+        UnsignedInteger = 0,
+        NegativeInteger,
+        ByteString,
+        TextString,
+        StartArray,
+        StartMap,
+        EndArray,
+        EndMap,
+        Tag,
+        Special,
+        Finished,
+        EndOfData,
```

Comment:
I really think that you should have EndOfData be the zero value.  It's the most likely to get successfully caught and debugged when used from a field that got zero-initted and not otherwise assigned.

Reply:
Perhaps then none of the cases should be assigned to zero?

Reply:
Well, the guidance is to always name the zero value... but to consider what the best zero value is.

"Unknown" is a reasonable "this is never valid" value, if you don't think EndOfData fits that role.

Reply:
Done